client.ts

```diff
@@ -11,31 +12,20 @@ export class Client {
         this.connectionParams = new ConnectionParams(config);
     }
 
-    async connect() {
+    async connect(): Promise<Connection> {
```

Comment:
@hayd can you revert this method as well as `query`? Let's make `Pool` new class. That way developers can use `Client` or `Pool` (`ClientPool`?)

Reply:
We might even have a common interface for both `Client` and `Pool`